### PR TITLE
ELPP-2775 Store ORCID access token

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: profiles/models.py [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
 from flask_sqlalchemy import SQLAlchemy
 
 db = SQLAlchemy()
 
 ID_LENGTH = 8
 
 
+class OrcidToken(db.Model):
+    orcid = db.Column(db.String(19), primary_key=True)
+    access_token = db.Column(db.String(255), unique=True)
```
Comment (giorgiosironi) [inline profiles/models.py]: why nullable on `access_token`?

File: profiles/repositories.py [hunk 2]
```diff
@@ -29,6 +39,21 @@ def next_id(self) -> str:
         raise NotImplementedError
 
 
+class SQLAlchemyOrcidTokens(OrcidTokens):
+    def __init__(self, db: SQLAlchemy) -> None:
+        self.db = db
```
Comment (giorgiosironi) [inline profiles/repositories.py]: why not injecting only `self.db.session`? It has a too short lifetime or will `self.db` soon be needed?

File: profiles/utilities.py [hunk 3]
```diff
@@ -1,5 +1,10 @@
 import random
 import string
+from datetime import datetime, timedelta
+
+
+def expires_at(expires_in: int) -> datetime:
```
Comment (giorgiosironi) [inline profiles/utilities.py]: don't understand why the spec put in a `expires_in` which makes the expiration time inaccurate while they could have provided a precise `expires_at`

File: profiles/repositories.py (hunk 2 above)
Comment (thewilkybarkid) [reply profiles/repositories.py]: Just following code seen elsewhere. Reading the docs it looks safe to inject `db.sessions`.

File: profiles/models.py (hunk 1 above)
Comment (thewilkybarkid) [reply profiles/models.py]: Good spot.

**Pull request merged**